src/main/java/attendance/AttendanceBook.java

```diff
@@ -0,0 +1,128 @@
+    return sorted(attendanceRecords).stream()
+        .collect(Collectors.toMap(
+            AttendanceRecord::getRecord,
+            AttendanceStatus::from,
+            (existing, replacement) -> existing,
+            LinkedHashMap::new
+        ));
+  }
+
+  public Map<AttendanceStatus, Integer> calculateAttendanceResult(String nickname,
+      LocalDate searchDate) {
+    Crew crew = new Crew(nickname);
+    validateExistentCrew(crew);
+    AttendanceRecords attendanceRecords = attendanceBook.get(crew);
+    AttendanceEvaluator attendanceEvaluator = new AttendanceEvaluator();
+    return attendanceEvaluator.calculateAttendanceResult(
+        attendanceRecords.searchRecords(searchDate));
+  }
+
+  public Map<Crew, ManagementStatus> checkManagementCrews(LocalDate searchDate) {
+    Map<Crew, ManagementStatus> checkedManagementCrewsResult = sortedManagementStatus(
+        searchDate).entrySet().stream()
+        .collect(Collectors.toMap(
+            Entry::getKey,
+            entry -> checkManagementCrew(entry.getValue()),
+            (existing, replacement) -> existing,
+            LinkedHashMap::new
+        ));
+    checkedManagementCrewsResult.entrySet()
+        .removeIf(entry -> entry.getValue().equals(ManagementStatus.GENERAL));
+    return checkedManagementCrewsResult;
+  }
+
+
+  public ManagementStatus checkManagementCrew(Map<AttendanceStatus, Integer> result) {
+    return ManagementStatus.from(applyAbsenceCountPolicy(result));
+  }
+
+  private Map<Crew, Map<AttendanceStatus, Integer>> sortedManagementStatus(LocalDate searchDate) {
+    return calculateCrewsAttendanceResult(searchDate).entrySet().stream()
+        .sorted(
+            Comparator.comparing(
+                    (Map.Entry<Crew, Map<AttendanceStatus, Integer>> entry) -> entry.getValue()
+                        .getOrDefault(AttendanceStatus.ABSENCE, Integer.MIN_VALUE),
+                    Comparator.reverseOrder())
+                .thenComparing(
+                    (Map.Entry<Crew, Map<AttendanceStatus, Integer>> entry) -> applyAbsenceCountPolicy(
+                        entry.getValue()), Comparator.reverseOrder())
+                .thenComparing(entry -> entry.getKey().getName())
+        )
+        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue,
+            (e1, e2) -> e1,
+            LinkedHashMap::new
+        ));
+  }
+
+  public Map<Crew, Map<AttendanceStatus, Integer>> calculateCrewsAttendanceResult(
+      LocalDate searchDate) {
+    return attendanceBook.entrySet().stream()
+        .collect(Collectors.toMap(
+                Map.Entry::getKey,
+                entry ->
+                    calculateAttendanceResult(entry.getKey().getName(), searchDate)
+            )
+        );
+  }
+
+  private int applyAbsenceCountPolicy(Map<AttendanceStatus, Integer> result) {
+    return result.get(AttendanceStatus.ABSENCE) + (result.get(AttendanceStatus.LATE)
+        / POLICY_CONDITION);
```

Comment:
```suggestion
    return result.get(AttendanceStatus.ABSENCE) + 
       (result.get(AttendanceStatus.LATE) / POLICY_CONDITION);
```
줄바꿈은 줄이 길어졌을때 시행하는 것도 있지만 어떻게 읽히느냐에 따라서 줄바꿈의 기준이 달라져야합니다. 다른 사람이 코드를 처음보더라도 한눈에 직관적이게 읽힐 수 있도록 해보세요.

Reply:
그럼 만약에 변수 명 또는 클래스 명이 너무 길어 선언하는 과정에서 줄바꿈이 되는 경우에는 아래처럼 표현하는 것도 좋은 방법일까요?

```
AttendanceRecord attendanceRecord = 
attendanceRecords.modifyAttendanceRecord(updateAttendanceRecord);
```

Reply:
현재 인텔리제이에서 기본으로 제공되는 컨벤션대로라면 줄바꿈이 내려가지 않을 것 같네요. 넘어가는 컨벤션이면 넘어가는 것이 맞습니다.